server/src/core/server.js

```diff
@@ -28,162 +27,133 @@ class Server {
+    log.success(`The current time zone is ${date.timeZone()}`)
 
-      if (!Object.keys(langs).includes(process.env.LEON_LANG) === true) {
-        process.env.LEON_LANG = 'en-US'
+    const sLogger = (process.env.LEON_LOGGER !== 'true') ? 'disabled' : 'enabled'
+    log.success(`Collaborative logger ${sLogger}`)
 
-        log.warning('The language you chose is not supported, then the default language has been applied')
-      }
-
-      log.success(`The current language is ${process.env.LEON_LANG}`)
-      log.success(`The current time zone is ${date.timeZone()}`)
-
-      const sLogger = (process.env.LEON_LOGGER !== 'true') ? 'disabled' : 'enabled'
-      log.success(`Collaborative logger ${sLogger}`)
-
-      await this.bootstrap()
-      resolve()
-    })
+    await Server.bootstrap()
   }
 
   /**
    * Bootstrap API
    */
-  static bootstrap () {
-    return new Promise(async (resolve) => {
-      const apiVersion = 'v1'
-
-      // Render the web app
-      app.use(express.static(`${__dirname}/../../../app/dist`))
-      app.get('/', (req, res) => {
-        res.sendFile(path.resolve(`${__dirname}/../../../app/dist/index.html`))
-      })
-
-      app.use(`/${apiVersion}/info`, infoRouter)
-      app.use(`/${apiVersion}/downloads`, downloadRouter)
-
-      try {
-        this.server = http.createServer(app)
-
-        await this.listen(process.env.LEON_PORT)
-        resolve()
-      } catch (e) {
-        log[e.type](e.obj.message)
-      }
+  static async bootstrap () {
+    const apiVersion = 'v1'
+
+    // Render the web app
+    fastify.register(fastifyStatic, {
+      root: path.join(__dirname, '..', '..', '..', 'app', 'dist'),
+      prefix: '/'
+    })
+    fastify.get('/', (_request, reply) => {
+      reply.sendFile('index.html')
     })
+
+    fastify.register(infoPlugin, { apiVersion })
+    fastify.register(downloadsPlugin, { apiVersion })
+    this.server = fastify.server
+    try {
+      await Server.listen(process.env.LEON_PORT)
+    } catch (e) {
+      log.error(e.message)
+    }
   }
 
   /**
    * Launch server
    */
-  static listen (port) {
-    return new Promise((resolve, reject) => {
-      const io = process.env.LEON_NODE_ENV === 'development'
-        ? socketio(this.server, { cors: { origin: `${process.env.LEON_HOST}:3000` } })
-        : socketio(this.server)
-
-      io.on('connection', this.connection)
-
-      this.server.listen(port, (err) => {
-        /* istanbul ignore if */
-        if (err) {
-          reject({ type: 'error', obj: err })
-          return
-        }
+  static async listen (port) {
+    const io = process.env.LEON_NODE_ENV === 'development'
+      ? socketio(this.server, { cors: { origin: `${process.env.LEON_HOST}:3000` } })
+      : socketio(this.server)
 
-        log.success(`Server is available at ${process.env.LEON_HOST}:${port}`)
+    io.on('connection', Server.connection)
 
-        resolve()
-      })
-    })
+    await fastify.listen(port, '0.0.0.0')
```

Comment:
Here we are assigning `0.0.0.0` to the `address` parameter of `fastify.listen`, because when we're using Docker, we need to assign this like this to make it work, it basically means, we're exposing on `localhost`, it should also work for others use case than Docker.

See: https://www.fastify.io/docs/latest/Getting-Started/#note